src/opendap/auth/UrsIdP.java

```diff
@@ -205,17 +243,134 @@ void getEDLUserProfile(UserProfile userProfile) throws IOException {
         userProfile.ingestJsonProfileString(contents);
     }
 
-// curl -X POST -d 'token=<token>&client_id=<‘your application client_id’> https://urs.earthdata.nasa.gov/oauth/tokens/user
+    // curl -X POST -d 'token=<token>&client_id=<‘your application client_id’>
+    // https://urs.earthdata.nasa.gov/oauth/tokens/user
+
+    /**
+     * TODO: add docstring? etc
+     * TODO: reconsider exception
+     * 
+     * @param accessToken
+     * @return
+     */
+    RSAPublicKey getPublicKeyForId(String publicKeyId) throws InvalidPublicKeyException, JsonSyntaxException {
+        // 1. Get the key set (JSON web key set, i.e. JWKS)
+        String jwksStr = getUrsClientPublicKeys();
+        JsonObject jwks = JsonParser.parseString(jwksStr).getAsJsonObject();
+        JsonArray jwksKeys = jwks.getAsJsonArray("keys");
+
+        // 2. Pull out the requested key
+        JsonObject jwkJson = null;
+        for (int i = 0; i < jwksKeys.size(); i++) {
+            JsonObject jwk = jwksKeys.get(i).getAsJsonObject();
+            if (jwk.has("kid")) {
+                if (jwk.get("kid").getAsString().equals(publicKeyId)) {
+                    jwkJson = jwk;
+                    break;
+                }
+            }
+        }
+        if (jwkJson == null) {
+            return null;
+        }
+
+        // 3. Convert key's json entry to a public key
+        Map<String, Object> jwkValues = new HashMap<>();
+        for (Map.Entry<String, JsonElement> entry : jwkJson.entrySet()) {
+            jwkValues.put(entry.getKey(), entry.getValue().getAsString());
+        }
+
+        try {
+            Jwk jwk = Jwk.fromValues(jwkValues);
+            return (RSAPublicKey) jwk.getPublicKey();
+        } catch (InvalidPublicKeyException e) {
+            log.error("Unable to get public key. Details: {}", e);
```

Comment:
This call, and others like it, are broken in that the parameter list doesn't match the format string. [The Logger methods debug, info, warn, error, and trace](https://www.slf4j.org/api/org/slf4j/Logger.html) (typically trace is not available) all have sadly complex method signatures and it's easy to run afoul of things:
```java
log.error(String, Throwable (aka Exception)) 
```
Does not expect or process a the format macro `{}` in the String.

You can drop the format macro `{} `and let the Logger work it out or be specific about what goes into the format macro:
```java
log.error("Unable to get public key. Details: {}", e.getMessage());
```